src/Utility/ExprFunctions.C

```diff
@@ -87,7 +92,56 @@ static void ExprException (const ExprEval::Exception& exc, const char* task,
 }
 
 
-int EvalFunc::numError = 0;
+/*!
+  \brief Static helper converting an index pair into a single index.
+  \details Assuming Voigt notation ordering; 11, 22, 33, 12, 23, 13.
+*/
+
+int voigtIdx (int d1, int d2)
+{
+  if (d1 > d2)
+    std::swap(d1,d2); // Assuming symmetry
+
+  if (d1 < 1 || d2 > 3)
+    return -1; // Out-of-range
+  else if (d2-d1 == 0) // diagonal term, 11, 22 and 33
+    return d1-1;
+  else if (d2-d1 == 1) // off-diagonal term, 12 and 23
+    return d2+1;
+  else // off-diagonal term, 13
+    return 5;
+}
```

Comment:
Since the `voigtIdx()` helper is only used once (in `addDerivative()`) it could as well be defined as a local Lamda in that method, just to make that more clear and since you here are moving the implementation anyway.

Reply:
ok.